tuxemon/core/states/world/worldstate.py

```diff
@@ -598,6 +604,30 @@ def get_explicit_tile_exits(self, position, tile, skip_nodes):
         except KeyError:
             pass
 
+    def tile_blocks(self, entity, tile):
+        """ Verifies if a collision tile is supposed to block a given type of NPC
+
+        Three checks are preformed:
+        If the NPC is a player, return True if "players" exists in the blocks
+        If the NPC isn't a player, return True if "npcs" exists in the blocks
+        If the slug of the NPC is present in the blocks, return True
+
+        :param entity: the player or NPC being checked
+        :param tile: the data of the tile being checked
+
+        :rtype: bool
+        """
+        if entity and tile and 'blocks' in tile:
+            for block in tile['blocks'].split():
+                if block == 'players' and entity.isplayer:
+                    return True
+                if block == 'npcs' and not entity.isplayer:
```

Comment:
we should not allow unexpected values in the map code.  if the only expected values are "players", "npcs" & a slug, then we should be raising an error here if the value is not valid.

Reply:
Correct, will fix in the next commit.

Reply:
There's actually a problem with this: I can't find a way to determine if a value is invalid here. The parameter may contain any number of NPC slugs, but the scan only occurs for the NPC trying to move onto that tile: It's impossible to check if the value might refer to the slug of another NPC and is thus value, as we're only working with the NPC moving in this call.

I could of course add a loop to scan all NPC's on the map and make sure the parameter matches at least one. But would the potential performance impact be worth it only to raise an exception for an incorrect value? If I can get a list of all map NPC's maybe I can use `if x in y:` which I understand is a lot cheaper so this might do.

Reply:
ok.  if i were doing this, i would make sure this block data is parsed when the map is loaded.  i would use the npc database and check if the slug is there, then check against the list ["players", "npcs"] and raise a ValueError if its not there.  the list would be checked again at runtime, but the database check is not needed at that time.

Reply:
Issue is that both the object I have access to (`self.npcs`) and the database would require a loop to extract the slugs. I can't directly do `if slug in npcs` as "npcs" will always contain an object and not a list of slugs. Only solution seems to be with this addition:

```
# create a list of npc slugs, used for checking parameter validity
npcs = []
for npc in self.get_all_entities():
    npcs.append(npc.slug)
```